src/server/licensing_func.c

```diff
@@ -624,6 +624,9 @@ license_nodes()
 				for (i = 0; i < np->nd_nummoms; i++)
 					propagate_licenses_to_vnodes(np->nd_moms[i]);
 			}
+		} else {
```

Comment:
Could you make use of remove_from_unlicensed_node_list() for this? Why do you need a unlicensed_node struct? Can't you add the link to original node strcture and get rid of unlicensed_node struct. So that you dont have to loop inside remove_from_unlicensed_node_list to find the node.

Reply:
As its a list of nodes to find that node in the list we have to traverse we cant be sure where and at what position desired node would be, so instead of using remove_from_unlicensed_node_list() I am directly removing the node from that list as we have already traversed above.

Reply:
I was proposing a change to remove_from_unlicensed_node_list so it does not have to traverse to remove the node from list
```
void
remove_from_unlicensed_node_list(struct pbsnode *pnode)
{
	if(!pnode->nd_added_to_unlicensed_list)
		return;

	pnode->nd_added_to_unlicensed_list = 0;
	licensing_control.licenses_total_needed -= pnode->nd_attr[ND_ATR_LicenseInfo].at_val.at_long;
	delete_link(&pnode->un_lic_link);
}
```

Reply:
I am not able to understand this change I think what we have now is good to go !

Reply:
@nithinj - are you proposing to remove the unlicensed_node_list itself.

Reply:
No, I was suggesting to use a link in pnode struct itself rather than in a separate struct like unlicensed_node

Reply:
I have made the changes @nithinj , please have a look on the latest commit